pkg/server/config.go

```diff
@@ -0,0 +1,418 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"reflect"
+
+	kcpadmissioninitializers "github.com/kcp-dev/kcp/pkg/admission/initializers"
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	kcpexternalversions "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	"github.com/kcp-dev/kcp/pkg/etcd"
+	"github.com/kcp-dev/kcp/pkg/indexers"
+	"github.com/kcp-dev/kcp/pkg/informer"
+	boostrap "github.com/kcp-dev/kcp/pkg/server/bootstrap"
+	"github.com/kcp-dev/kcp/pkg/server/options"
+	"github.com/kcp-dev/kcp/pkg/server/requestinfo"
+	"github.com/kcp-dev/logicalcluster"
+	etcdtypes "go.etcd.io/etcd/client/pkg/v3/types"
+	apiextensionsapiserver "k8s.io/apiextensions-apiserver/pkg/apiserver"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apiextensionsexternalversions "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/endpoints/filters"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/apiserver/pkg/util/webhook"
+	"k8s.io/client-go/dynamic"
+	coreexternalversions "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clusters"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/genericcontrolplane"
+	"k8s.io/kubernetes/pkg/genericcontrolplane/apis"
+)
+
+type tokenAuth struct {
+	kcp       string
+	shard     string
+	shardHash []byte
+}
+type Config struct {
+	options                    *options.CompletedOptions
+	EtcdServer                 *etcd.Server
+	identityConfig             *rest.Config
+	resolveIdentities          func(ctx context.Context) error
+	genericConfig              *genericapiserver.Config
+	apisConfig                 *apis.Config
+	apiExtensionsConfig        *apiextensionsapiserver.Config
+	preHandlerChain            handlerChainMuxes
+	dynamicClusterClient       *dynamic.Cluster
+	apiextensionsClusterClient *apiextensionsclient.Cluster
+	kcpClusterClient           *kcpclient.Cluster
+	kubeClusterClient          *kubernetes.Cluster
```

Comment:
this list needs some better grouping. A block of client related things would make sense. The sub-apiserver config should go to the top.